src/NodeCompiler/CompilerContext.php

```diff
@@ -5,41 +5,59 @@
 namespace Roave\BetterReflection\NodeCompiler;
 
 use Roave\BetterReflection\Reflection\ReflectionClass;
+use Roave\BetterReflection\Reflection\ReflectionFunctionAbstract;
 use Roave\BetterReflection\Reflector\Reflector;
 use RuntimeException;
 
+/**
+ * @internal
+ */
 class CompilerContext
 {
-    public function __construct(private Reflector $reflector, private ?ReflectionClass $self)
+    public function __construct(private Reflector $reflector, private ?string $fileName, private ?string $namespace, private ?ReflectionClass $class, private ?ReflectionFunctionAbstract $function)
     {
     }
 
-    /**
-     * Does the current context have a "self" or "this"
-     *
-     * (e.g. if the context is a function, then no, there will be no self)
-     */
-    public function hasSelf(): bool
+    public function getReflector(): Reflector
+    {
+        return $this->reflector;
+    }
+
+    public function getFileName(): ?string
+    {
+        return $this->fileName;
+    }
+
+    public function getNamespace(): ?string
     {
-        return $this->self !== null;
+        return $this->namespace;
     }
 
-    public function getSelf(): ReflectionClass
+    public function inClass(): bool
```

Comment:
Overall no big deal with this design, since it's `@internal`, but we should probably avoid `hasFoo()` and `getFoo` design when possible. A union type with an enumeration or such would be preferable in future.

Reply:
I'm happy to remove this methods. I used the pattern just because it was there... Give me 10 minutes :)

Reply:
No need to, just keep it in mind for your future machinations 😁

Reply:
https://github.com/Roave/BetterReflection/pull/766 ;)